scripts/host-check

```diff
@@ -1412,7 +1451,7 @@ def main(argv: List[str]) -> NoReturn:
         success = run_specified_checks(args.platform, args.extra_checks)
     else:
         # We've already checked that extra checks aren't specified in this case.
-        success = run_all_checks()
+        success = run_all_checks(args.extra_checks)
```

Comment:
"run_all_checks" and "run_specified_checks" no longer seem like such good names - would it make sense to combine the logic into a single function? Or just rename?

Reply:
Just renamed (and sorted out the doc strings) - there's enough output that's a bit different whether you've specified a platform or not that it's a bit awkward to get it into a single function.